### PR TITLE
DegreePlanner: add `planner_list_all` command

### LINKED ISSUE #72: As a user, I want to list my degree planners
... so that I can have an overview of my current plan.

### PR DESCRIPTION
**Pull request by Letm3through**
Resolves #72 

Since the current application does not have:

- a feature to list all degree planners
- UI to display degree planners

Let's add and update the followings to do so:
* [ 1/10] [View: add UI for DegreePlannerList](https://github.com/CS2113-AY1819S2-T09-1/main/pull/81/commits/ee53f2e28ed2d409bf640d93fd70af17ae0c77df)
* [ 2/10] [View: update MainWindow to include UI for DegreePlannerList](https://github.com/CS2113-AY1819S2-T09-1/main/pull/81/commits/9a72a08bff36a6ff824804da881b9e322df9fd42)
* [ 3/10] [Commands: add PlannerListAllCommand](https://github.com/CS2113-AY1819S2-T09-1/main/pull/81/commits/47758ff6b7bbfb465af4448f97a21097b7ae7076)
* [ 4/10] [Logic: update to include PlannerListAllCommand](https://github.com/CS2113-AY1819S2-T09-1/main/pull/81/commits/5c2f004db8d6208d14e9efb57c16a63247ca718c)
* [ 5/10] [UI: add DegreePlanner related classes to handle UI for DegreePlannerList](https://github.com/CS2113-AY1819S2-T09-1/main/pull/81/commits/212fe202396bd14c74472a7bc9956b9b5a7ed430)
* [ 6/10] [UI: update MainWindow to handle DegreePlannerList](https://github.com/CS2113-AY1819S2-T09-1/main/pull/81/commits/08b44ff07b094095b80782ddd7f06002e99e3abe)
* [ 7/10] [View: update BrowserPanel to be hidden](https://github.com/CS2113-AY1819S2-T09-1/main/pull/81/commits/f05e85ee0bd8b2071984fc999c1bdd9bb50655c9)
* [ 8/10] [Test: update to include PlannerListAllCommand](https://github.com/CS2113-AY1819S2-T09-1/main/pull/81/commits/9fbc0e5d9810048b6695f2cf075c9467e7512353)
* [ 9/10] [Test: update HelpCommandSystemTest to not click on browserPannel](https://github.com/CS2113-AY1819S2-T09-1/main/pull/81/commits/483a9fdcc3aa16431da953c11c0d84ead88808f8)
* [10/10] [Util: update SampleDegreePlannerUtil to include various modules](https://github.com/CS2113-AY1819S2-T09-1/main/pull/81/commits/d584c0becc0d3469e48737b1ba174b77456b8714)

Disclaimer: 
UI test for ```DegreePlannerList``` will be updated after:

-  removal of ```BrowserPanel``` along with the removal of test for ```BrowserPanel```
- Update of model and storage structures for DegreePlannerList

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/PlannerListAllCommand.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        requireNonNull(model);
+        String degreePlannerListContent = "";
+        for (DegreePlanner degreePlanner : model.getFilteredDegreePlannerList()) {
+            degreePlannerListContent +=
+                    "Year: " + degreePlanner.getYear() + "\tSemester: " + degreePlanner.getSemester() + "\n";
+            if (degreePlanner.getCodes().isEmpty()) {
+                degreePlannerListContent += "No module inside";
+            } else {
+                degreePlannerListContent +=
+                        "Modules: " + degreePlanner.getCodes().toString().replaceAll("\\[|\\]", " ");
```
Comment (Creastery) [inline src/main/java/seedu/address/logic/commands/PlannerListAllCommand.java]: Please avoid such string manipulations.
You should be looping through the `Set<Code>` and concatenating them.

File: src/main/java/seedu/address/logic/commands/PlannerListAllCommand.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+public class PlannerListAllCommand extends Command {
+
+    public static final String COMMAND_WORD = "planner_list_all";
+
+    public static final String MESSAGE_SUCCESS = "Listed all degree planners:\n";
+
+    @Override
+    public CommandResult execute(Model model, CommandHistory history) {
+        requireNonNull(model);
+        String degreePlannerListContent = "";
```
Comment (Creastery) [inline src/main/java/seedu/address/logic/commands/PlannerListAllCommand.java]: StringBuilder would have been more appropriate here.

File: src/main/java/seedu/address/logic/commands/PlannerListAllCommand.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import seedu.address.model.planner.DegreePlanner;
+
+/**
+ * Lists all degree planners in the address book to the user.
+ */
+public class PlannerListAllCommand extends Command {
+
+    public static final String COMMAND_WORD = "planner_list_all";
+
+    public static final String MESSAGE_SUCCESS = "Listed all degree planners:\n";
```
Comment (Creastery) [inline src/main/java/seedu/address/logic/commands/PlannerListAllCommand.java]: This should be a format string!

File: src/main/java/seedu/address/logic/commands/PlannerListAllCommand.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                    "Year: " + degreePlanner.getYear() + "\tSemester: " + degreePlanner.getSemester() + "\n";
+            if (degreePlanner.getCodes().isEmpty()) {
+                degreePlannerListContent += "No module inside";
+            } else {
+                degreePlannerListContent +=
+                        "Modules: " + degreePlanner.getCodes().toString().replaceAll("\\[|\\]", " ");
+            }
+            degreePlannerListContent += "\n\n";
+        }
+        return new CommandResult(MESSAGE_SUCCESS + degreePlannerListContent);
```
Comment (Creastery) [inline src/main/java/seedu/address/logic/commands/PlannerListAllCommand.java]: You should be avoiding this. Rather, `MESSAGE_SUCCESS` should be a format string, and you can supply the `String` to format the message.

File: src/main/java/seedu/address/model/util/SampleDegreePlannerUtil.java [hunk 5]
```diff
@@ -18,21 +18,21 @@
     public static DegreePlanner[] getSampleDegreePlanners() {
         return new DegreePlanner[] {
             new DegreePlanner(new Year("1"), new Semester("1"),
-                    getCodeSet()),
+                    getCodeSet("CS1010", "CS1231", "MA1101R", "GER1000", "EC1301")),
```
Comment (Creastery) [inline src/main/java/seedu/address/model/util/SampleDegreePlannerUtil.java]: Avoid adding `Code` that do not exists in `SampleDegreePlannerUtil`.

File: src/main/java/seedu/address/ui/DegreePlannerCard.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+    private Label semester;
+
+    @FXML
+    private ListView<String> degreePlannerListView;
+
+    public DegreePlannerCard(DegreePlanner degreePlanner) {
+        super(FXML);
+        this.degreePlanner = degreePlanner;
+
+        year.setText("YEAR: " + degreePlanner.getYear().year);
```
Comment (Creastery) [inline src/main/java/seedu/address/ui/DegreePlannerCard.java]: "YEAR" -> "Year".

File: src/main/java/seedu/address/ui/DegreePlannerCard.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+        semester.setText("\tSEM: " + degreePlanner.getSemester().plannerSemester);
```
Comment (Creastery) [inline src/main/java/seedu/address/ui/DegreePlannerCard.java]: "SEM" -> "Semester" and replace `\t` with a space.

**Review by Creastery — CHANGES_REQUESTED** [review]
Please see review comments. Thanks!

File: src/main/java/seedu/address/model/util/SampleDegreePlannerUtil.java (hunk 5 above)
Comment (Letm3through) [reply src/main/java/seedu/address/model/util/SampleDegreePlannerUtil.java]: Noted and updated. Thanks!

**Review by Creastery — APPROVED** [review]

**Pull request merged**